.github/workflows/publishWheelRelease.yml

```diff
@@ -0,0 +1,129 @@
+      LLVM_BUILD_DIR: "llvm-build"
+      # The setup.py script expects the StableHLO build to be in 'build'
+      STABLEHLO_BUILD_DIR: "build"
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ "3.10", "3.11" ]
+    runs-on: ${{ github.repository == 'openxla/stablehlo' && 'ubuntu-22.04-64core' ||  'ubuntu-22.04'  }}
+
+    steps:
+    - name: Checkout StableHLO
+      uses: actions/checkout@v4
+      # TODO(fzakaria): We need all the commits and the tags for the
+      # git describe` command to work to find the correct version.
+      # This can be removed if the version can be determmined in another way.
+      with:
+        fetch-depth: 0
+    - name: Get LLVM Version
+      id: llvm-version
+      shell: bash
+      run: |
+        echo "version=$(cat ${{ github.workspace }}/build_tools/llvm_version.txt)" >> $GITHUB_OUTPUT
+
+    - name: Setup workspace
+      uses: ./.github/actions/setup-build
+      with:
+        llvm-version: ${{ steps.llvm-version.outputs.version }}
+        python-version: ${{ matrix.python-version }}
+
+    # The actual tool need to execute the python build
+    # TODO(fzakaria): Consider using cibuildwheel to build multiple wheels at
+    # once and mark them as manylinux compatible.
+    - run: pip install build
+
+    - name: Configure and Build LLVM
+      shell: bash
+      run: |
+        ./build_tools/github_actions/ci_build_cmake_llvm.sh "$LLVM_PROJECT_DIR" "$LLVM_BUILD_DIR"
+      env:
+          CMAKE_BUILD_TYPE: Release
+          MLIR_ENABLE_BINDINGS_PYTHON: ON
+
+    - name: Build and Test StableHLO with Python
+      shell: bash
+      run: |
+        ./build_tools/github_actions/ci_build_cmake.sh "$LLVM_BUILD_DIR" "$STABLEHLO_BUILD_DIR"
+      env:
+          CMAKE_BUILD_TYPE: Release
+          STABLEHLO_ENABLE_BINDINGS_PYTHON: ON
+
+    - name: Build Python Wheel
+      shell: bash
+      run: |
+        cd "$GITHUB_WORKSPACE/stablehlo/integrations/python"
+        python3 -m build -w --outdir $GITHUB_WORKSPACE/${{ github.sha }}
+
+    - name: Upload an artifact
+      uses: actions/upload-artifact@v4
+      with:
+        if-no-files-found: error
+        name: build_artifact-${{ matrix.python-version }}
+        path: ${{ github.sha }}
+
+  upload-tarballs:
+    runs-on: ubuntu-20.04
+    needs: cmake-build
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ "3.10", "3.11" ]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Download artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: build_artifact-${{ matrix.python-version }}
+          path: ${{ github.sha }}
+
+      - name: Release current commit
+        uses: ncipollo/release-action@v1.13.0
+        with:
+          artifacts: "${{ github.sha }}/*.whl"
+          token: "${{ secrets.GITHUB_TOKEN }}"
+          # git describe in setup.py explicitly filters out this tag
+          tag: "dev-wheels"
```

Comment:
We tag the current release as dev-wheels for the GitHub release.
This name is then explicitly filtered out via `git describe` which is somewhat cursed.